runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -131,21 +131,55 @@ public void deleteTable(String namespace, String tableName) {
     }
 
     /**
-     * Start appending mutations to a transaction.
-     * The transaction does not begin until a commit is invoked.
-     * On a commit the latest available snapshot will be used to resolve the transaction.
+     * Start a transaction at the isolation level of the latest available corfu snapshot.
+     * The transaction does not begin until either a commit is invoked or a read happens.
      *
      * @param namespace Namespace of the tables involved in the transaction.
      * @return Returns a transaction builder instance.
      */
     @Nonnull
+    @Deprecated
     public TxBuilder tx(@Nonnull final String namespace) {
         return new TxBuilder(
                 this.runtime.getObjectsView(),
                 this.runtime.getTableRegistry(),
                 namespace);
     }
 
+    /**
+     * Start a transaction at the isolation level of the latest available corfu snapshot.
+     * The transaction does not begin until either a commit is invoked or a read happens.
+     *
+     * @param namespace Namespace of the tables involved in the transaction.
+     * @return Returns a transaction builder instance.
+     */
+    @Nonnull
+    public TxnContext txn(@Nonnull final String namespace) {
+        return new TxnContext(
+                this.runtime.getObjectsView(),
+                this.runtime.getTableRegistry(),
+                namespace,
+                IsolationLevel.snapshot());
+    }
+
+    /**
+     * Start appending mutations to a transaction.
+     * The transaction does not begin until either a commit or the first read is invoked.
+     * On read or commit the latest available snapshot will be used to resolve the transaction.
+     *
+     * @param namespace Namespace of the tables involved in the transaction.
+     * @param isolationLevel Snapshot (latest or specific) at which the transaction must execute.
+     * @return Returns a transaction builder instance.
+     */
+    @Nonnull
+    public TxnContext txn(@Nonnull final String namespace, IsolationLevel isolationLevel) {
```

Comment:
not sure I understand this interface...  Is it to be used for read_committed isolation level in future?  How is it different from the previous method?

Reply:
Backward compatibility. tx() method returns the old implementation to avoid changing existing consumers.